numba/np/arraymath.py

```diff
@@ -437,177 +439,117 @@ def array_var_impl(arr):
             ssd += np.real(val * np.conj(val))
         return ssd / arr.size
 
-    res = context.compile_internal(builder, array_var_impl, sig, args)
-    return impl_ret_untracked(context, builder, sig.return_type, res)
+    return array_var_impl
 
 
-@lower_builtin(np.std, types.Array)
-@lower_builtin("array.std", types.Array)
-def array_std(context, builder, sig, args):
-    def array_std_impl(arry):
-        return arry.var() ** 0.5
-    res = context.compile_internal(builder, array_std_impl, sig, args)
-    return impl_ret_untracked(context, builder, sig.return_type, res)
+@overload(np.std)
+@overload_method(types.Array, "std")
+def array_std(x):
+
+    def array_std_impl(x):
+        return x.var() ** 0.5
+
+    return array_std_impl
 
 
 def zero_dim_msg(fn_name):
     msg = ("zero-size array to reduction operation "
-           "{0} which has no identity".format(fn_name))
+           f"{fn_name} which has no identity")
     return msg
 
 
-@lower_builtin(np.min, types.Array)
-@lower_builtin("array.min", types.Array)
-def array_min(context, builder, sig, args):
-    ty = sig.args[0].dtype
-    MSG = zero_dim_msg('minimum')
-
-    if isinstance(ty, (types.NPDatetime, types.NPTimedelta)):
-        def array_min_impl(arry):
-            if arry.size == 0:
-                raise ValueError(MSG)
+@overload(np.min)
+@overload_method(types.Array, "min")
+def npy_min(x):
+    if isinstance(x.dtype, (types.NPDatetime, types.NPTimedelta)):
+        pre_return_func = njit(lambda x: np.isnat(x))
+        comparator = njit(lambda x, min_val: x < min_val)
```

Comment:
It might be a good idea to hoist the comparator code into a `@register_jitable` code and use this function instead. This way you can avoid redefining the same comparator in other functions.